apps/vr-tests-react-components/src/stories/CustomStyleHooks.stories.tsx

```diff
@@ -57,7 +68,11 @@ export const CompoundButtonCustomStyles = () => {
   const customStyleHooks: FluentProviderCustomStyleHooks = {
     useCompoundButtonStyles_unstable: (state: unknown) => {
       const componentState = state as CompoundButtonState;
-      componentState.root.className = mergeClasses(componentState.root.className, styles.button);
+      componentState.root.className = mergeClasses(
```

Comment:
Should we be publicly documenting this in the docsite? As we discussed in tech sync, we should be using this to unblock advanced usage. The API we settled on is a good compromise to reduce the risk of abuse.

I would rather go the direction of - if someone knows to use this without asking us, then they probably know what they are doing. If we get reported this issue in the future we can either:

1. point to a long term solution
2. explain how to use this hack and that it shouldn't be used for any other use case

Reply:
yeah, happy to leave the story in the repo for reference, but not publish to docsite.

Reply:
> Should we be publicly documenting this in the docsite? As we discussed in tech sync, we should be using this to unblock advanced usage. The API we settled on is a good compromise to reduce the risk of abuse.
> 
> I would rather go the direction of - if someone knows to use this without asking us, then they probably know what they are doing. If we get reported this issue in the future we can either:
> 
> 1. point to a long term solution
> 2. explain how to use this hack and that it shouldn't be used for any other use case

@ling1726 I wouldn't necessarily call this advanced usage. At least, it's not more advanced than any of the rest of the custom styling hook API. In fact, it's (currently) required to use this function to correctly implement a custom style hook.

Reply:
I'll pull out the public docsite change to another PR, so this one can be unblocked for now.

Reply:
* #34266